### ISSUE TITLE
Apigee Emulator VsCode extension Target Servers SSL Issue

### ISSUE THREAD

**Issue by dmngu9** [GoogleCloudPlatform/cloud-code-vscode#595]
**Environment**:
local environment with Apigee Emulator . Using Cloud Code extension on VsCode

**Version information**

Cloud Code Extension version: 1.17.2

VSCode version: 1.52.1

OS: MacOS Catalina 10.15.7

**Description**:
I set up a target endpoint as shown below. The target server is public Postman mock server

```
<HTTPTargetConnection>
        <LoadBalancer>
            <Server name="targetA" />
        </LoadBalancer>
</HTTPTargetConnection>
```

targetservers.json

```
[
  {
    "enabled": true,
    "name": "targetA",
    "host": "{host}",
    "port": 443,
    "sSLInfo": {
      "enabled": "true"
    }
  }
]
```

When i make a request , i got this error in response

```
{
    "fault": {
        "faultstring": "Unexpected EOF at target",
        "detail": {
            "errorcode": "messaging.adaptors.http.flow.UnexpectedEOFAtTarget"
        }
    }
}
```

However, it does work if port is 80 (http)

My question is does Apigee Emulator and Cloud Code VSCode extension support target server with SSL? I have a feeling the target server does not support sSLInfo

**Comment by shirishtrivikr** [comment]
That is correct we are working on getting full TLS/SSLInfo support for target servers.

**Comment by velsietis** [comment]
Is there an ETA on getting this issue resolved?
It really reduces the usefulness of the emulator if you we can't use Target Servers (many of our backend targets are only available over HTTPS)

**Comment by shirishtrivikr** [comment]
Yes this will be available soon. This requires changes in the emulator as well the extension.  When both the changes are rolled out we will update the ticket with the information.

**Comment by velsietis** [comment]
> Yes this will be available soon. This requires changes in the emulator as well the extension. When both the changes are rolled out we will update the ticket with the information.

Thanks great to hear.  Thanks @shirishtrivikr for the update!

**Comment by erica-addepar** [comment]
Are here any updates on this? I'm using 1.8.0 of the Apigee Emulator

**Comment by juliensie** [comment]
hey @erica-addepar, I'm using `1.9.0` and there is an issue with `443` and `HTTPS`

**Comment by shirishtrivikr** [comment]
@juliensie 
Thanks for your feedback.  Is there a specific use case that does not work for you?
Could I please request you to raise a separate issue which details the issue you are seeing.

**Comment by juliensie** [comment]
hi @shirishtrivikr

I've just added
 
```xml
<SSLInfo>
      <Enabled>true</Enabled>
</SSLInfo>
```
in `<HTTPTargetConnection>` and it works now 😃

**Comment by juliensie** [comment]
Like @dmngu9, 
if I don't add `<SSLInfo>` element in my target endpoint definition, adding 
```json
"port": 443,
"sSLInfo": {
      "enabled": "true"
}
```
is **not enough** and Request to target endpoint will be `http` instead of `https` 😢

**Comment by velsietis** [comment]
> hi @shirishtrivikr
> 
> I've just added
> 
> ```
> <SSLInfo>
>       <Enabled>true</Enabled>
> </SSLInfo>
> ```
> 
> in `<HTTPTargetConnection>` and it works now 😃

@juliensie - is that when using Target Servers in the configuration (i.e. the `<Server name="targetA" />` element rather than a `<URL>https://example.com</URL>` element?

This issue is specifically about the TargetServers not supporting SSL in the emulator (see the original comment:  https://github.com/GoogleCloudPlatform/cloud-code-vscode/issues/595#issue-1259463459).

I'm assuming that still doesn't work as the issue is still open...

**Comment by juliensie** [comment]
Sure using `<Server>` element in `<LoadBalancer>` element, like this:
```xml
<LoadBalancer>
      <Server name="targetA"/>
</LoadBalancer>
```

No problem using `https` protocole in `<URL>` element so far.

**Comment by shirishtrivikr** [comment]
@dmngu9 @erica-addepar 

The full TLS and key store support is now available as a part of the insider release. You would need to use the 1.9.1 emulator as a prerequisite.

Here are relevant documentation links.
[Configuring the key stores](https://cloud.google.com/apigee/docs/api-platform/local-development/vscode/deploy-environment#key-stores)
[Configuring the target servers](https://cloud.google.com/apigee/docs/api-platform/local-development/vscode/deploy-environment#target-servers)

**Comment by kurtkanaskie** [comment]
I'm still not seeing this work, I'm using the insiders build and running 1.9.1.
My target server that works in X is:
```
{
  "name": "pingstatus-v1",
  "host": "xapi-dev.myhost.net",
  "port": 443,
  "isEnabled": true,
  "sSLInfo": {
    "enabled": true
  },
  "protocol": "HTTP"
}
```
My Target endpoint is:
```
<HTTPTargetConnection>
    <!-- Not OK -->
    <LoadBalancer>
        <Server name="pingstatus-v1"/>
    </LoadBalancer>
    <Path>{flow.target.basepath}/{flow.target.pathsuffix}</Path>
    <!-- OK
    <URL>https://xapi-dev.myhost.net/pingstatus-mock/v1/{flow.target.pathsuffix}</URL>
    -->
    <!-- OK
    <URL>https://xapi-dev.myhost.net/{flow.target.basepath}/{flow.target.pathsuffix}</URL>
    -->
</HTTPTargetConnect
```
I'm getting the `UnexpectedEOFAtTarget` fault.

**Comment by shirishtrivikr** [comment]
@kurtkanaskie The target servers are included in emulator 1.9.2 could you please try that ?

**Comment by kurtkanaskie** [comment]
Hmmm, just tried, still seeing `UnexpectedEOFAtTarget`
